File: LeetCode/Unknown/백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <string>

using namespace std;

int main() {
    string str, bomb;
    cin >> str >> bomb;

    string answer = "";
    for (char c : str) {
        answer += c;

        if (answer.size() >= bomb.size()) {
            bool flag = true;
            for (int j = 0; j < bomb.size(); j++) {
                if (answer[answer.size() - bomb.size() + j] != bomb[j]) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                for (int j = 0; j < bomb.size(); j++) {
                    answer.pop_back();
                }
            }
        }
    }

    if (answer.empty()) cout << "FRULA";
    else cout << answer;

    return 0;
}

File: LeetCode/Unknown/백준/Gold/9935. 문자열 폭발/문자열 폭발.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] arg) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String str, bomb;
        str = br.readLine();
        bomb = br.readLine();

        StringBuilder answer = new StringBuilder();
        for (int i = 0; i < str.length(); i++) {
            answer.append(str.charAt(i));

            if (answer.length() >= bomb.length()) {
                boolean flag = true;
                for (int j = 0; j < bomb.length(); j++) {
                    if (answer.charAt(answer.length() - bomb.length() + j) != bomb.charAt(j)) {
                        flag = false;
                        break;
                    }
                }
                if (flag) {
                    answer.setLength(answer.length() - bomb.length());
                }
            }
        }

        if (answer.length() == 0) {
            System.out.println("FRULA");
        }
        System.out.println(answer);
    }

}
